great_expectations/execution_engine/sqlalchemy_batch_data.py

```diff
@@ -291,4 +302,119 @@ def _create_temporary_table(  # noqa: C901, PLR0912
                 self.execution_engine.execute_query_in_transaction(sa.text(stmt_2))
         else:
             self.execution_engine.execute_query_in_transaction(sa.text(stmt))
-        return stmt
+        return (stmt, temp_table_name)
+
+    def _generate_selectable_from_schema_name_and_table_name(
+        self,
+        dialect: GXSqlDialect,
+        use_quoted_name: bool,
+        table_name: str,
+        schema_name: Optional[str] = None,
+    ) -> sqlalchemy.Table:
+        """Helper method to generate selectable using schema and table name
+
+        Args:
+            dialect (GXSqlDialect): Needed to check for BigQuery, which needs to be handled differently.
+            use_quoted_name (bool): To be passed to sqlalchemy.
+            table_name (str): Table name to build selectable from.
+            schema_name (Optional[str], optional): Optional schema name. Defaults to None.
+
+        Returns:
+            sqlalchemy.Table: SqlAlchemy Table that is Selectable.
+        """
+        if use_quoted_name:
+            table_name = sqlalchemy.quoted_name(table_name, quote=True)
+        if dialect == GXSqlDialect.BIGQUERY:
+            if schema_name is not None:
+                logger.warning(
+                    "schema_name should not be used when passing a table_name for biquery. Instead, include the schema name in the table_name string."
+                )
+            # In BigQuery the table name is already qualified with its schema name
+            return sa.Table(
+                table_name,
+                sa.MetaData(),
+                schema=None,
+            )
+        return sa.Table(
+            table_name,
+            sa.MetaData(),
+            schema=schema_name,
+        )
+
+    def _generate_selectable_from_query(
+        self,
+        query: str,
+        dialect: GXSqlDialect,
+        create_temp_table: bool,
+        temp_table_schema_name: Optional[str] = None,
+    ) -> sqlalchemy.Table:
+        """Helper method to generate Selectable from query string.
+
+        Args:
+            query (str): query passed in as RuntimeBatchRequest.
+            dialect (GXSqlDialect): Needed for _create_temporary_table, since different backends name temp_tables differently.
+            create_temp_table (bool): Should we create a temp_table?
+            temp_table_schema_name (Optional[str], optional): Optional string for temp_table schema.  Defaults to None.
+
+        Returns:
+            sqlalchemy.Table: SqlAlchemy Table that is Selectable.
+        """
+        if not create_temp_table:
+            return sa.text(query)
+        (_, temp_table_name) = self._create_temporary_table(
```

Comment:
```suggestion
        _, temp_table_name = self._create_temporary_table(
```
??

Reply:
Thanks for the catch here @Kilo59. I made the change in one more location for the file too 🙇🏼